packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -332,6 +332,20 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     markNeedsTextLayout();
   }
 
+  /// Character used for obscuring text if [obscureText] is true.
+  ///
+  /// Cannot be null, and must have a length of exactly one.
+  String get obscuringCharacter => _obscuringCharacter;
+  String _obscuringCharacter;
+  set obscuringCharacter(String value) {
+    if (_obscuringCharacter == value) {
+      return;
+    }
+    assert(value != null && value.length == 1);
```

Comment:
nit: is it necessary to impose the `length == 1` constraint in the render object? I thought only `EditableTextState` makes assumption about the length of the obscuring string. For example `obscuringCharacter = ''` seems to be a legit use case here in case someone wants to hide the length of the password.

Reply:
I feel the empty character is a bit weird here. how would the selection or cursor work in this case?

Reply:
Yeah then they would have to implement their own `TextInputClient` etc.

Reply:
I'll keep an eye out for anybody asking for this kind of feature.